common/network/queues/concurrent_queue.c.inc

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdlib.h>
+
+#define DEFINE_VTABLE_CONCURRENT_QUEUE_OF(TYPE) \
+  vtable_concurrent_queue_of_##TYPE##_t const   \
+      vtable_concurrent_queue_of_##TYPE##_g =   \
+          {CONCURRENT_QUEUE_OF_##TYPE##_FRONT,  \
+           CONCURRENT_QUEUE_OF_##TYPE##_BACK,   \
+           CONCURRENT_QUEUE_OF_##TYPE##_EMPTY,  \
+           CONCURRENT_QUEUE_OF_##TYPE##_SIZE,   \
+           CONCURRENT_QUEUE_OF_##TYPE##_PUSH,   \
+           CONCURRENT_QUEUE_OF_##TYPE##_POP};
+
+#define NEW_CONCURRENT_QUEUE_OF(TYPE, NAME)                            \
+  struct concurrent_queue_of_##TYPE##_s *NAME = malloc(sizeof(*NAME)); \
```

Comment:
Any potential memory leak resulting from this allocation?

Reply:
The user has the responsibility to free the queue